### PR TITLE
Admin Pannel Design Changes/Challenge Dependencies

### PR DESCRIPTION
**Pull request by Merrixx**
#### Basic Requirements

- [X] I have added Documentation https://github.com/CTF-Citadel/ctf-citadel.github.io/pull/36
- [X] I have tested my code
- [X] I have synced my branch with main

#### Describe your PR, what does it fix/add?

Changes Design for Event Time Selection (Makes it look better)
Challenges can depend on other Challenges
Challenges with child Challenges cannot be deleted

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Dependencies checking needs to be implemented into Challenge listing by checking the corresponding attribute

### CODE REVIEW

File: src/lib/actions.ts [hunk 1]
```diff
@@ -173,7 +184,8 @@ class DatabaseActions {
             challenge_difficulty: diff,
             needs_container: isContainer,
             container_file: filePath,
-            static_file_url: fileURL
+            static_file_url: fileURL,
+            depends_on: dependon
```
Comment (0xk1f0) [inline src/lib/actions.ts:188]: dependOn

File: src/lib/actions.ts [hunk 2]
```diff
@@ -322,15 +334,15 @@ class DatabaseActions {
...
+    async updateChallenge(id: string, name: string, desc: string, cat: string, diff: string, event: string, children: string) {
         await DB_ADAPTER.update(challenges)
             .set({
                 challenge_name: name,
                 challenge_description: desc,
                 challenge_category: cat,
                 challenge_difficulty: diff,
                 event_id: event
-            })
+                })
```
Comment (0xk1f0) [inline src/lib/actions.ts:345]: ?

File: src/lib/backend.ts [hunk 3]
```diff
@@ -193,8 +193,9 @@ export async function privilegedWrapper(request: Request): Promise<Response> {
                     json.data.isContainer,
                     json.data.filePath,
                     json.data.fileURL,
-                    json.data.event
-                );
+                    json.data.event,
+                    json.data.dependon
+                    );
```
Comment (0xk1f0) [inline src/lib/backend.ts:198]: formatting

File: src/lib/actions.ts [hunk 4]
```diff
@@ -162,7 +172,8 @@ class DatabaseActions {
         isContainer: boolean,
         filePath: string,
         fileURL: string,
-        toEvent: string
+        toEvent: string,
+        dependon: string,
```
[the remark this answers is not in the document]
Comment (Merrixx) [reply src/lib/actions.ts:176]: Changed

File: src/lib/actions.ts (hunk 1 above)
Comment (Merrixx) [reply src/lib/actions.ts:188]: Changed

File: src/lib/actions.ts (hunk 2 above)
Comment (Merrixx) [reply src/lib/actions.ts:345]: Formatting

File: src/lib/backend.ts (hunk 3 above)
Comment (Merrixx) [reply src/lib/backend.ts]: Done

File: src/components/AdminPanel.svelte [hunk 5]
```diff
@@ -134,6 +137,12 @@
         const DATA = await requestWrapper(true, { type: 'challenges' });
         const JSON = await DATA.json();
         challenges = JSON.data;
+        selectchallenges = [];
+        challenges.forEach((element: any) => {
+            selectchallenges.push({ value: element.id, name: element.challenge_name });
+        });
+        console.log(challenges);
+        console.log(selectchallenges);
```
Comment (0xk1f0) [inline src/components/AdminPanel.svelte:145]: remove pls

File: src/components/AdminPanel.svelte [hunk 6]
```diff
@@ -226,6 +235,21 @@
         } else return false;
     }
 
+    async function checkChildDepends(dependAmount: any) {
+        disableDependDelete = false;
+        console.log(disableDependDelete);
```
Comment (0xk1f0) [inline src/components/AdminPanel.svelte:240]: remove

**Review by 0xk1f0 — CHANGES_REQUESTED** [review]

File: src/components/AdminPanel.svelte (hunk 5 above)
Comment (Merrixx) [reply src/components/AdminPanel.svelte]: Done

File: src/components/AdminPanel.svelte (hunk 6 above)
Comment (Merrixx) [reply src/components/AdminPanel.svelte]: Done

**Review by 0xk1f0 — APPROVED** [review]

**Pull request merged**